src/Agent.Worker/TaskCommandExtension.cs

```diff
@@ -511,6 +515,62 @@ private void ProcessTaskSetTaskVariableCommand(IExecutionContext context, Dictio
             context.TaskVariables.Set(name, data, isSecret);
         }
 
+        private void ProcessTaskSetEndpointCommand(IExecutionContext context, Dictionary<string, string> eventProperties, string data)
+        {
+            String endpointIdInput;
+            if (!eventProperties.TryGetValue(TaskSetEndpointEventProperties.EndpointId, out endpointIdInput) || String.IsNullOrEmpty(endpointIdInput))
+            {
+                throw new Exception(StringUtil.Loc("MissingEndpointID"));
+            }
+
+            Guid endpointId;
+            if (!Guid.TryParse(endpointIdInput, out endpointId))
+            {
+                throw new Exception(StringUtil.Loc("InvalidEndpointId"));
+            }
+
+            String key;
+            if (!eventProperties.TryGetValue(TaskSetEndpointEventProperties.Key, out key) || String.IsNullOrEmpty(key))
+            {
+                throw new Exception(StringUtil.Loc("MissingEndpointKey"));
+            }
+
+            var endpoint = context.Endpoints.Find(a => a.Id == endpointId);
+            if (EqualityComparer<ServiceEndpoint>.Default.Equals(endpoint, default(ServiceEndpoint)))
+            {
+                throw new Exception(StringUtil.Loc("InvalidEndpointId"));
+            }
+
+            String field;
+            if (!eventProperties.TryGetValue(TaskSetEndpointEventProperties.Field, out field) || String.IsNullOrEmpty(field))
+            {
+                throw new Exception(StringUtil.Loc("MissingEndpointField"));
+            }
+
+            if(String.Equals(field, "dataParameter", StringComparison.OrdinalIgnoreCase))
+            {
+                endpoint.Data[key]=data;
+            }
+            else if(String.Equals(field, "authParameter", StringComparison.OrdinalIgnoreCase))
+            {
+                endpoint.Authorization.Parameters[key]= data;
```

Comment:
you need to also update the secret masker about the new value for the auth. parameter. need to add the value being set as a mask like below:
                var _secretMasker = HostContext.GetService<ISecretMasker>();
                _secretMasker.AddRegex(data);

take a look at ProcessTaskSetSecretCommand as reference 

it seems fine to mask the auth. parameter unconditionally for now as there is no context reg. whether auth. parameter is confidential or not in the agent. i think this is safer than task setting whether auth parameter is secret or not when invoking setendpoint command.